pkg/solana/codec/solana.go

```diff
@@ -0,0 +1,297 @@
+/*
+Package codec provides functions to create a codec from an Anchor IDL using NewIDLCodec. All primitives map 1:1, however
+some common types are either not supported or may not have expected behavior.
+
+Using time.Time as a native struct for encoding or decoding to and IDL unixTimestamp primitive will not work without
+the codec modifier `ModifierEpochToTime`. The native value to represent timestamps is int64.
+
+Enums are only supported in their basic form of uint8 values. Enums with variants are not supported.
+*/
+package codec
```

Comment:
Will you need new modifiers to extract fields?  Since you get back the whole account, the part you want to read may be a subfield.